linera-views/src/key_value_store_view.rs

```diff
@@ -124,6 +137,11 @@ where
                     Update::Set(value) => batch.put_key_value_bytes(key, value),
                 }
             }
+            do_something
+        };
+        if do_something {
+            let key = self.context.base_tag(KeyTag::Sizes as u8);
+            batch.put_key_value(key, &self.key_sizes)?;
         }
         let hash = *self.hash.get_mut();
         if self.stored_hash != hash {
```

Comment:
Do we need to clear `self.key_sizes` in `clear` below, too?

Reply:
Yes, and it is worse than that since we need also to do for the `rollback` which means that we need to store the preceding value of the `key_size`.